src/scripts/start_valheim.sh

```diff
@@ -63,6 +63,56 @@ cp /home/steam/steamcmd/linux64/steamclient.so /home/steam/valheim/linux64/
 log "Initializing Variables...."
 odin configure || exit 1
 
+log "Checking for TYPE flag"
+export TYPE="${TYPE:="vanilla"}"
+log "Found Type ${TYPE}"
+export TYPE="${TYPE,,}"
+export GAME_LOCATION="${GAME_LOCATION:="/home/steam/valheim"}"
+
+
+if \
+  # ValheimPlus not yet installed
+  { [ "${TYPE}" = "valheimplus" ] && [ ! -d "${GAME_LOCATION}/BepInEx" ] && [ ! -f "${GAME_LOCATION}/BepInEx/plugins/ValheimPlus.dll" ]; } || \
+  # ValheimPlus with update on startup or force install
+  { [ "${TYPE}" = "valheimplus" ] && { [ "${UPDATE_ON_STARTUP:-0}" -eq 1 ] || [ "${FORCE_INSTALL:-0}" -eq 1 ]; } ; }; then
+    log "Installing ValheimPlus"
+    VALHEIM_PLUS_URL="$(curl https://api.github.com/repos/valheimPlus/ValheimPlus/releases/latest | jq -r '.assets[] | select(.name=="UnixServer.zip") | .browser_download_url')"
+    log "Pulling ValheimPlus from ${VALHEIM_PLUS_URL}"
+    odin installmod "${VALHEIM_PLUS_URL}"
+elif \
+  # BepInEx not yet installed
+  { [ "${TYPE}" = "bepinex" ] && [ ! -d "${GAME_LOCATION}/BepInEx" ] && [ ! -f "${GAME_LOCATION}/BepInEx/core/BepInEx.dll" ]; } || \
+  # BepInEx with update on startup or force install
+  { [ "${TYPE}" = "bepinex" ] && { [ "${UPDATE_ON_STARTUP:-0}" -eq 1 ] || [ "${FORCE_INSTALL:-0}" -eq 1 ]; } ; }; then
+    log "Installing BepInEx"
+    BEPINEX_URL="$(curl https://valheim.thunderstore.io/api/experimental/package/denikson/BepInExPack_Valheim/ | jq -r '.latest.download_url')"
+    log "Pulling BepInEx from ${BEPINEX_URL}"
+    odin installmod "${BEPINEX_URL}"
+elif \
+  # BepInEx not yet installed
+  { [ "${TYPE}" = "bepinexfull" ] && [ ! -d "${GAME_LOCATION}/BepInEx" ] && [ ! -f "${GAME_LOCATION}/BepInEx/core/BepInEx.dll" ]; } || \
+  # BepInEx with update on startup or force install
+  { [ "${TYPE}" = "bepinexfull" ] && { [ "${UPDATE_ON_STARTUP:-0}" -eq 1 ] || [ "${FORCE_INSTALL:-0}" -eq 1 ]; } ; }; then
+    log "Installing BepInEx Full"
+    BEPINEX_URL="$(curl https://valheim.thunderstore.io/api/experimental/package/1F31A/BepInEx_Valheim_Full/ | jq -r '.latest.download_url')"
+    log "Pulling BepInEx Full from ${BEPINEX_URL}"
+    odin installmod "${BEPINEX_URL}"
+fi
+
+log "Running with ${TYPE} Valheim <3"
+
+if [ "${TYPE}" = "valheimplus" ] || [ "${TYPE}" = "bepinex" ]  || [ "${TYPE}" = "bepinexfull" ]; then
```

Comment:
Can we add in a warning or error if `$TYPE` is vanilla and `$MODS` is set to something that the mods won't actually be installed without one of the supported mod frameworks?

Reply:
Yee